Sources/CodeEditSourceEditor/Enums/CaptureModifier.swift

```diff
@@ -0,0 +1,133 @@
+//
+//  CaptureModifiers.swift
+//  CodeEditSourceEditor
+//
+//  Created by Khan Winter on 10/24/24.
+//
+
+// https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#semanticTokenModifiers
+
+/// A collection of possible syntax capture modifiers. Represented by an integer for memory efficiency, and with the
+/// ability to convert to and from strings for ease of use with tools.
+///
+/// These are useful for helping differentiate between similar types of syntax. Eg two variables may be declared like
+/// ```swift
+/// var a = 1
+/// let b = 1
+/// ```
+/// ``CaptureName`` will represent both these later in code, but combined ``CaptureModifier`` themes can differentiate
+/// between constants (`b` in the example) and regular variables (`a` in the example).
+///
+/// This is `Int8` raw representable for memory considerations. In large documents there can be *lots* of these created
+/// and passed around, so representing them with a single integer is preferable to a string to save memory.
+///
+public enum CaptureModifier: Int8, CaseIterable, Sendable {
+    case declaration
+    case definition
+    case readonly
+    case `static`
+    case deprecated
+    case abstract
+    case async
```

Comment:
Does async need to go in backticks?
```suggestion
    case `async`
```

Reply:
It doesn't need backticks to compile and I can't think of why except that `async` can only exist in function definitions, so it's different from something like `extension` or `func`

Reply:
Awesome, I kind of figured as much. I just wanted to throw that out there just in case.